### PR TITLE
docs(AUTHORS.rst): Polish contents.

### PR DESCRIPTION
**Pull request by aureooms**
- Remove duplicated content.
  - Update some dead links.
  - Use "secure" URLs only.
  - Add details.

### CODE REVIEW

File: AUTHORS.rst [hunk 1]
```diff
@@ -41,18 +40,17 @@ Many thanks for your contributions!
 There are also a few modules or functions incorporated from other
 authors and projects:
 
-* ``utils.iso_precision`` includes something like 10 lines of Arrow's, code,
-  which is written by Chris Smith and under Apache license
+* The function ``ics.utils.iso_precision`` includes something like 10 lines of
```
Comment (N-Coder) [inline AUTHORS.rst:43]: We can probably drop this paragraph entirely, as the `iso_precision` function is gone together with arrow (we now get the precision directly from reading the `VALUE` type parameter).

File: AUTHORS.rst [hunk 2]
```diff
@@ -41,18 +40,17 @@ Many thanks for your contributions!
...
+A big part of the code was written at `UrLab <https://urlab.be>`_, an awesome
+hackerspace at `ULB <https://ulb.be>`_ in Brussels, Belgium.
 
 Thanks to:
 
-* `arrow <http://crsmithdev.com/arrow/>`_ which provides a nice API for dates, times and deltas,
-* `python-dateutil <http://labix.org/python-dateutil>`_ for parsing timezones from VTIMEZONE blocks,
-* `requests <http://python-requests.org/>`_ for giving me inspiration from it's beautiful and pythonic API (and its doc too),
-* `six <http://pythonhosted.org/six/>`_ to have made the python3 transition easier,
+* `arrow <https://arrow.readthedocs.io/en/latest>`_ which provides a nice API for dates, times and deltas,
```
Comment (N-Coder) [inline AUTHORS.rst:52]: Maybe also drop this line, as the API turned out to be not-so-nice for floating datetimes (i.e. without timezone, which is explicitly allowed by the spec). For this reason, `arrow` was completely removed and we're now using plain but correct `datetimes`. See #222.

File: AUTHORS.rst [hunk 3]
```diff
@@ -41,18 +40,17 @@ Many thanks for your contributions!
...
+* `python-dateutil <https://dateutil.readthedocs.io/en/stable>`_ for parsing timezones from VTIMEZONE blocks,
```
Comment (N-Coder) [inline AUTHORS.rst:53]:
```suggestion
* `python-dateutil <https://dateutil.readthedocs.io/en/stable>`_ for parsing VTIMEZONEs and RRULEs,
```
RRULE support should be coming with #240 (see [here](https://github.com/C4ptainCrunch/ics.py/pull/240/files#diff-aaf03cd716cb3f30c8f8172b896c9719R49)).
Maybe we should also include Tatsu in this list? (#194)

File: AUTHORS.rst (hunk 1 above)
Comment (aureooms) [reply AUTHORS.rst:43]: Yes!

File: AUTHORS.rst (hunk 2 above)
Comment (aureooms) [reply AUTHORS.rst:52]: Yes!

File: AUTHORS.rst (hunk 3 above)
Comment (aureooms) [reply AUTHORS.rst:53]: Yes!

File: AUTHORS.rst [hunk 4]
```diff
@@ -32,7 +32,6 @@ Other contributors, listed alphabetically, are:
 * `@Timic3 <https://github.com/Timic3>`_
 * `@tomschr <https://github.com/tomschr>`_
 * `@Trii <https://github.com/Trii>`_
-* `@tomschr <https://github.com/tomschr>`_
```
Comment (aureooms) [inline AUTHORS.rst:35]: Nice catch!

File: AUTHORS.rst (hunk 2 above)
Comment (C4ptainCrunch) [reply AUTHORS.rst:52]:
```suggestion
```

File: AUTHORS.rst (hunk 1 above)
Comment (C4ptainCrunch) [reply AUTHORS.rst:43]:
```suggestion
```

**Pull request merged**